core/src/main/java/io/stargate/core/util/ByteBufferUtils.java

```diff
@@ -47,6 +55,21 @@ public static ByteBuffer fromBase64UrlParam(String base64) {
     return ByteBuffer.wrap(Base64.getUrlDecoder().decode(base64));
   }
 
+  private static boolean isLegacyEncodedBase64String(String base64) {
+    // From
+    // https://cowtowncoder.medium.com/measuring-string-indexofany-string-performance-java-fecb9eb473fa
```

Comment:
since we are re-implementing it, we should add a unit test for this new implementation.

Reply:
The logic is covered by existing unit tests; I verified that the legacy case is covered as well as regular one.